src/compiler/scala/tools/nsc/classpath/ZipAndJarFileLookupFactory.scala

```diff
@@ -73,7 +73,9 @@ object ZipAndJarClassPathFactory extends ZipAndJarFileLookupFactory {
     override private[nsc] def classes(inPackage: PackageName): Seq[ClassFileEntry] = files(inPackage)
 
     override protected def createFileEntry(file: FileZipArchive#Entry): ClassFileEntryImpl = ClassFileEntryImpl(file)
-    override protected def isRequiredFileType(file: AbstractFile): Boolean = file.isClass
+    override protected def isRequiredFileType(file: AbstractFile, fileExists: String => Boolean): Boolean = {
+      file.isClass && !(file.hasExtension("class") && fileExists(file.name.dropRight(6) + ".tasty"))
```

Comment:
So `dirEntry.entries` contains just file names, not paths? Should we change the function name to `siblingExists`?

Reply:
that's right, the keys are only the names - so yeah that change would make sense